### PR TITLE
FDP-109 Add distributed throttling service

### PR DESCRIPTION
**Pull request by bvdzwet**
Adds a Spring Boot application that offers REST endpoints that can be
called from several distributed clients to maintain maximum concurrent
connections per segment constraints accessing a network that is
segmented by base transceiver stations with cells.

### CODE REVIEW

**Comment by Marcel-Jansen** [comment]
retest this please

**Comment by bvdzwet** [comment]
retest this please

File: osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/cleanup/PermitCleanUpJob.java [hunk 1]
```diff
@@ -35,35 +39,66 @@
...
-          }
-        });
+    if (this.batchSize < 1) {
+      LOGGER.warn(
+          "Not cleaning any permits, as batch size is configured at a value less than one: cleanup.permits.batch.size={}",
+          this.batchSize);
+      return;
+    }
+    final Instant createdAtBefore = Instant.now().minus(this.timeToLive);
+    final PageRequest pageRequest = PageRequest.of(0, this.batchSize, Sort.by("id").ascending());
```
Comment (sanderv) [inline osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/cleanup/PermitCleanUpJob.java:54]: https://www.baeldung.com/jpa-limit-query-results#1first-ortop

Comment (sanderv) [reply osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/cleanup/PermitCleanUpJob.java:54]: Sorting doesn't seem to add anything here, might even impact query time. Just get the first _x_ permits.

**Pull request merged**